package/src/downsampler_worker_handler.ts

```diff
@@ -0,0 +1,127 @@
+/*
+    Copyright 2018-2022 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+import { DownsamplerWorkerRequest } from './types';
+import Downsampler from './downsampler';
+
+let downsampler: Downsampler | null = null;
+onmessage = async function(event: MessageEvent<DownsamplerWorkerRequest>): Promise<void> {
+  switch (event.data.command) {
+    case 'init':
+      if (downsampler !== null) {
+        self.postMessage({
+          command: 'error',
+          message: 'Downsampler already initialized',
+        });
+        return;
+      }
+      try {
+        Downsampler.setWasm(event.data.wasm);
+        downsampler = await Downsampler.create(
+          event.data.inputSampleRate,
+          event.data.outputSampleRate,
+          event.data.filterOrder,
+          event.data.frameLength,
+        );
+        self.postMessage({
+          command: 'ok',
+          version: downsampler.version,
+        });
+      } catch (e: any) {
+        self.postMessage({
+          command: 'error',
+          message: e.message,
+        });
+      }
+      break;
+    case 'process':
+      if (downsampler === null) {
+        self.postMessage({
+          command: 'error',
+          message: 'Downsampler not initialized',
+        });
+        return;
+      }
+      try {
+        const { inputFrame } = event.data;
+        let outputBuffer = new Int16Array(inputFrame.length);
```

Comment:
this is fine for now, but if we moved to new `resampler` the logic here falls apart. There is a function `pv_resampler_convert_num_samples_to_output_sample_rate` that let you know an upper bound of the outputBuffer size based on the length of the inputFrame.